exchanges/bitfinex/bitfinex_test.go

```diff
@@ -478,34 +478,35 @@ func TestUpdateTickers(t *testing.T) {
 	t.Parallel()
 
 	err := b.UpdateTradablePairs(context.Background(), false)
-	if err != nil {
-		t.Fatal(err)
-	}
+	assert.NoErrorf(t, err, "UpdateTradablePairs should not error")
 
 	assets := b.GetAssetTypes(false)
-	for x := range assets {
+	for _, a := range assets {
 		var avail []currency.Pair
-		avail, err = b.GetAvailablePairs(assets[x])
-		if err != nil {
-			t.Fatal(err)
-		}
-
-		err = b.CurrencyPairs.StorePairs(assets[x], avail, true)
-		if err != nil {
-			t.Fatal(err)
-		}
-
-		err = b.UpdateTickers(context.Background(), assets[x])
-		if err != nil {
-			t.Fatal(err)
-		}
-
-		for y := range avail {
-			_, err = ticker.GetTicker(b.Name, avail[y], assets[x])
-			if err != nil {
-				t.Fatal(err)
+		avail, err = b.GetAvailablePairs(a)
+		assert.NoError(t, err, "GetAvailablePairs should not error")
+
+		err = b.CurrencyPairs.StorePairs(a, avail, true)
+		assert.NoError(t, err, "StorePairs should not error")
+
+		err = b.UpdateTickers(context.Background(), a)
+		assert.NoError(t, err, "UpdateTickers should not error")
+
+		// Bitfinex leaves delisted pairs in Available info/conf endpoints
+		// We want to assert that most pairs are valid, so we'll check that no more than 5% are erroring
+		acceptableThreshold := 0.95
+		okay := 0.0
+		var errs error
+		for _, p := range avail {
+			if _, err = ticker.GetTicker(b.Name, p, a); err != nil {
+				errs = common.AppendError(errs, err)
+			} else {
+				okay++
 			}
 		}
+		if !assert.Greater(t, okay, acceptableThreshold*float64(len(avail)), "No more than %d%% of %s tickers should error", int(acceptableThreshold*100), a) {
```

Comment:
No more than 95% of spot tickers should error, so maybe change threshold to 5% and maybe swap the assert length check around? 

Also locally its failing due to that memecoin Bonk wigging out: `Bitfinex BONK-UST bid equals ask this is a crossed or locked market`

Reply:
Revised to make the error saner:
```
    bitfinex_test.go:513:
                Error Trace:    /Users/gbjk/go/src/github.com/gbjk/gocryptotrader/exchanges/bitfinex/bitfinex_test.go:513
                Error:          "63.58381502890174" is not greater than "95"
                Test:           TestUpdateTickers
                Messages:       At least 95% of spot tickers should not error
```

Fixed cf31bcbe

Reply:
ExcludeError to the rescue again.
Fixed 599d9c0f

However: This introduces an actual functional issue as well.
Bitfinex is bailing on the Update tickers with the failure to BONK.
I fixed this somewhere else too so I'll cargo-cult it in now. Two dog-seconds and I'll have it in.

Reply:
Fixed 8eb02bf8

PR details updated